### DISCUSSION TITLE
Bug in using count

### DISCUSSION THREAD

**Discussion by iamB0rgy** [spatie/laravel-searchable#137] [Q&A]
This 
`        $term = $request->input('search');
        $searchResults = (new Search())
            ->registerModel(User::class, 'username', 'name', 'profession')
            ->registerModel(Recipe::class, 'recipe_name')
            ->perform($term);`

Works with `$searchResults->count()`

But this, 

`        $searchResults = (new Search())
            ->registerModel(User::class, function (ModelSearchAspect $userSearchAspect) use ($term) {
            $userSearchAspect
            ->addSearchableAttribute('name')
            ->orWhereHas('profileDetail', function (Builder $query, $term) {
                $query->whereRaw('LOWER(profession) like %'  . mb_strtolower($term, 'UTF8') . '%');
            })

            ->perform($term);
        });`

doesn't work with `$searchResults->count()`. 
Error is `Call to undefined method Spatie\Searchable\Search::count()`

**Comment by nolsen89** [comment]
i have the same problem, when i try to use the function attribute

**Reply by biswajitpaul01** [reply]
Do we have a solution for this? 🙂
